addons/api/addon/models/role.js

```diff
@@ -42,6 +51,61 @@ export default class RoleModel extends GeneratedRoleModel {
       : null;
   }
 
+  /**
+   * Convenience for looking up the grant scopes, if loaded.
+   */
+  get grantScopes() {
+    // filter out scopes that start with an o_ or p_
+    // then sort the keywords
+    const grantScopes = [];
+    if (this.grant_scope_ids) {
+      const keywordIDs = this.grant_scope_ids.filter((id) => {
+        return !id.startsWith('o_') && !id.startsWith('p_') && id !== 'global';
+      });
+      // using filter instead of find to get the global id and allow for
+      // spreading into sortedScopeIDs if globalID is undefined
+      const globalID = this.grant_scope_ids.filter((id) => {
+        return id === 'global';
+      });
+      const orgIDs = this.grant_scope_ids.filter((id) => {
+        return id.startsWith('o_');
+      });
+      const projectIDs = this.grant_scope_ids.filter((id) => {
+        return id.startsWith('p_');
+      });
+
+      const sortedKeywords = keywordIDs.sort((a, b) => {
+        return (
+          GRANT_SCOPE_KEYWORDS.indexOf(a) - GRANT_SCOPE_KEYWORDS.indexOf(b)
```

Comment:
Maybe make a comment on the array that the order is important and affects sorting if we're doing this

Reply:
added